### ISSUE TITLE
Add Show Output Functionality to SAM CLI

### ISSUE THREAD

**Issue by amcquistan** [aws/aws-sam-cli#2813]
### Describe your idea/feature/enhancement

I have often thought it would be nice to be able to use the SAM CLI to reshow the output from my SAM/CloudFormation stack to lookup things like the HTTP endpoint to a Api Gateway resource to testing or sharing with other developers / end users.

For example, I can spin up the pre-canned hello world Python template like so.

```
samdev init --name sam-show-output-poc \
  --runtime python3.6 \
  --app-template hello-world \
  --package-type Zip
```

Then build & deploy and I get this very nice output for values specified in the output section.

```
cd sam-show-output-poc
samdev build --use-container
samdev deploy --stack-name sam-show-output-poc --region us-east-2 --guided
```

I'll omit / skip to only the output I'm interested in for this explanation.

```
CloudFormation outputs from deployed stack
-----------------------------------------------------------------------------------------------------------------------------
Outputs
-----------------------------------------------------------------------------------------------------------------------------
Key                 HelloWorldFunctionIamRole
Description         Implicit IAM Role created for Hello World function
Value               arn:aws:iam::accountid:role/sam-show-output-poc-HelloWorldFunctionRole-somesuffix

Key                 HelloWorldApi
Description         API Gateway endpoint URL for Prod stage for Hello World function
Value               https://6s4m40aq2l.execute-api.us-east-2.amazonaws.com/Prod/hello/

Key                 HelloWorldFunction
Description         Hello World Lambda Function ARN
Value               arn:aws:lambda:us-east-2:accountid:role:function:sam-show-output-poc-HelloWorldFunction-somesuffix
-----------------------------------------------------------------------------------------------------------------------------

Successfully created/updated stack - sam-show-output-poc in us-east-2
```

if I ever want to easily look up the HelloWorldApi URL I must login to the AWS console and click my into the CloudFormation views and then to the output tab of the "sam-show-output-poc" stack to see this again.

Instead, I'd like to be able to do the following to get the output section of my CF stack.

```
samdev show --stack-name sam-show-output-poc
```

### Proposal

I am proposing adding some variant of a `sam show` or perhaps `sam output` command that will show the stack output section from the CLI. The majority of the functionality for this is already present in the code base just need a little refactoring and reorganizing to expose it in a reusable way for the existing `sam deploy` command and shared use in this new `sam show` functionality.

### Additional Details

Please see initial PR / POC work here: https://github.com/aws/aws-sam-cli/pull/2812

**Comment by jfuss** [comment]
@amcquistan Thanks for the proposal. This seems pretty similar to https://github.com/aws/aws-sam-cli/issues/2770 but solved in a different manner. 

What is the benefit of this being in SAM CLI vs using CloudFormation API's?

**Comment by specious** [comment]
`aws cloudformation describe-stacks` will issue AWS API calls and retrieve the outputs generated during the last deployment.

If you add or change an output definition in the template file, then it won't take effect until the next actual deployment.

I agree that it would be great to be able to see the outputs at any time using sam.

**Comment by specious** [comment]
`aws cloudformation describe-stacks` will issue AWS API calls and retrieve the outputs generated during the last deployment.

If you add or change an output definition in the template file, then it won't take effect until the next actual deployment.

I agree that it would be very convenient to be able to see the outputs at any time using sam.

**Comment by specious** [comment]
> "... its a bit awkward / verbose and I can never exactly remember the command until I try it a few times."

In the [Litter Map](https://github.com/littermap) project, we include an administrative toolkit that [makes it easy](https://github.com/littermap/littermap-aws-backend/blob/1b16f497497bf44da07823314ee00da6207942a6/manage#L198-L202) to see the stack outputs from the last deployment by running `./manage list-stack-outputs`.

Feel free to borrow code (it's published under GPLv3).

**Comment by mildaniel** [comment]
This feature has now been released in version 1.72.0!
